pkg/certify/external-bash/driver-call.go

```diff
@@ -0,0 +1,342 @@
+package externalBash
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"github.com/wongma7/csi-certify/pkg/certify/utils"
+	"k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+	"k8s.io/kubernetes/test/e2e/storage/testsuites"
+	"os/exec"
+	"strings"
+)
+
+const (
+	getDriverInfo = "getDriverInfo"
+	createVolume  = "createVolume"
+)
+
+var RunCustomTestDriver = true
+var scriptName = ""
+var currentNameSpace = "default"
+var currentDriver *bashDriver
+
+var preprovisionedVolumeTestDriver = false
+var preprovisionedPVTestDriver = false
+
+type BashDriverParameter struct {
+}
+
+type testVolume struct {
+	volumeAttrib map[string]string
+}
+
+var bashDriverParam BashDriverParameter
+
+func init() {
+	flag.Var(&bashDriverParam, "bash-testdriver", "name of bashscript that implements all the required testdriver functions")
+}
+
+func (b BashDriverParameter) String() string {
+	return "<bash testdriver file>"
+}
+
+func (b BashDriverParameter) Set(filename string) error {
+	RunCustomTestDriver = false
+	scriptName = filename
+
+	validTestDriver := checkBashFuncExists("getDriverInfo")
+	if validTestDriver == false {
+		return errors.Errorf("Invalid TestDriver, must include getDriverInfo() function")
+	}
+
+	preprovisionedVolumeTestDriver = checkBashFuncExists("createVolume") && checkBashFuncExists("deleteVolume")
+	preprovisionedPVTestDriver = preprovisionedVolumeTestDriver
```

Comment:
make these 2 a property of bashDriver, then you can do driver.preprovisionedVolumeTestDriver = blah after you call getDriverDefinition